adminpage/views.py

```diff
@@ -1,3 +1,272 @@
+                raise LogicError('Cannot modify bookEnd time of a activity started')
+
+        if int(time.mktime(activity.book_start.timetuple())) < int(time.time()):
+            if activity.total_tickets != self.input['totalTickets']:
+                raise LogicError('Cannot modify totalTickets after book_start')
+
+        activity.name = self.input['name']
+        activity.place = self.input['place']
+        activity.description = self.input['description']
+        activity.pic_url = self.input['picUrl']
+        activity.start_time = (datetime.strptime(self.input['startTime'], "%Y-%m-%dT%H:%M:%S.%fZ") + timedelta(hours=8))
+        activity.end_time = (datetime.strptime(self.input['endTime'], "%Y-%m-%dT%H:%M:%S.%fZ") + timedelta(hours=8))
+        activity.book_start = (datetime.strptime(self.input['bookStart'], "%Y-%m-%dT%H:%M:%S.%fZ") + timedelta(hours=8))
+        activity.book_end = (datetime.strptime(self.input['bookEnd'], "%Y-%m-%dT%H:%M:%S.%fZ") + timedelta(hours=8))
+        activity.total_tickets = self.input['totalTickets']
+        activity.status = self.input['status']
+
+        activity.save()
+
+
+class ActivityMenu(APIView):
+
+    def get(self):
+        if not self.request.user.is_authenticated():
+            raise ValidateError('User has not logged in.')
+        activityListAll = Activity.objects.all()
+        wechat_menu = CustomWeChatView.lib.get_wechat_menu()
+        if len(wechat_menu) >= 2:
+            actiListInMenu = wechat_menu[1]['sub_button']
+        else:
+            actiListInMenu = []
+        data = []
+        for acti in activityListAll:
+            if acti.status == Activity.STATUS_DELETED:
+                continue
+            temp = {
+                'id': acti.id,
+                'name': acti.name,
+                'menuIndex': 0
+            }
+            i = 0
+            while i < len(actiListInMenu):
+                if actiListInMenu[i]['name'] == acti.name:
+                    temp['menuIndex'] = i+1
+                    break
+                i += 1
+
+            data.append(temp)
+        return data
+
+    def post(self):
+        if not self.request.user.is_authenticated():
+            raise ValidateError('User has not logged in.')
+        activityList = []
+        for activity_id in self.input:
+            activity = Activity.get_by_id(activity_id)
+            activity.status = Activity.STATUS_PUBLISHED
+            activity.save()
+            activityList.append(activity)
+        CustomWeChatView.update_menu(activityList)
+
+
+class ImageUpload(APIView):
+
+    def post(self):
+        self.check_input('image')
+        if not self.request.user.is_authenticated():
+            raise ValidateError('User has not logged in.')
+        image = self.input['image'][0]
+        try:
+            with open('static/media/img/' + image.name, 'wb') as img_file:
```

Comment:
the name of pic should be identical, it has the risk of covering original picture